integration-tests/tests/run-catalog-filter.js

```diff
@@ -38,12 +39,27 @@ const testCatalogFilterFn = withUserToken(async ({ testUrl, page, token }) => {
     throw new Error(`Rows not filtered after searching for '${searchText}'`)
   }
 
+  // Testing filter by facet
   await click(page, clickable({ text: filterItem, isDescendant: true }))
   const datasetSizeAfterFilter = await getDatasetCount(page)
 
   if (datasetSizeAfterFilter >= datasetSizeAfterSearch) {
     throw new Error(`Filter for '${filterItem}' was not applied to the table`)
   }
+  
+  // Testing filter by multiple same facets
+  await click(page, clickable({ text: 'Controlled', isDescendant: true }))
+  const datasetSizeAfterFilter2 = await getDatasetCount(page)
+  if (datasetSizeAfterFilter2 !== 0) {
+    throw new Error(`Filters for 'Controlled' should be ANDed between the same facet category in the table'`)
+  }
+
+  // Testing clearing filters
+  await click(page, clickable({ textContains: 'clear' }))
+  const datasetSizeAfterClear = await getDatasetCount(page)
+  if (datasetSizeAfterClear !== datasetSizeAfterSearch) {
```

Comment:
Should this throw an error if `datasetSizeAfterClear !== totalDatasetSize` instead?

Reply:
The text filter is still applied at this point of the unit tests, so this test matches the behavior of the application. I guess the question we should ask is whether or not the clear button should also clear out the search text too?

Reply:
Oh I see! Then in the error message, should it include the `datasetSizeAfterSearch` instead of the `totalDatasetSize`?

Reply:
> I guess the question we should ask is whether or not the clear button should also clear out the search text too?

Good point! Maybe @lstrano-broad has some insight?